app/jobs/thumbnail_job.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class ThumbnailJob < ApplicationJob
+  def perform(web_resource_id, ore_aggregation_association)
+    @web_resource_id = web_resource_id
+    aggregation = ORE::Aggregation.find_by("#{ore_aggregation_association}._id": BSON::ObjectId.from_string(web_resource_id))
+    case ore_aggregation_association
+    when 'edm_hasViews'
+      web_resource = aggregation.send(ore_aggregation_association.to_sym).select { |wr| wr._id == @web_resource_id }.first
+    when 'edm_isShownBy'
+      web_resource = aggregation.send(ore_aggregation_association.to_sym)
+    else
+      raise
+    end
+    @uploader = web_resource.media
+    %w(200x200 400x400).each do |size|
```

Comment:
This, and much of the logic in this job, belongs in `MediaUploader` which should utilise [Carrierwave's version and processing support](https://github.com/carrierwaveuploader/carrierwave#adding-versions).

[Conditional versioning](https://github.com/carrierwaveuploader/carrierwave#conditional-versions) should be used to prevent this happening during the upload & save process, and it would then be the primary role of this Sidekiq job to instruct `MediaUploader` to generate its thumbnails.